src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -34,7 +34,7 @@ internal partial class Binder
             // Create a map from type parameter name to ordinal.
             // No need to report duplicate names since duplicates
             // are reported when the type parameters are bound.
-            var names = new Dictionary<string, int>(n);
+            var names = new Dictionary<string, int>(n, StringOrdinalComparer.Instance);
```

Comment:
Yes. default comparer will eventually do ordinal compare, but first it jumps through some hoops figuring what default comparer corresponds to the given type, where it is located, touches some static generic fields.

This change just makes it easier to get the comparer without any indirections.